network/common/src/main/java/org/apache/spark/network/protocol/MessageWithHeader.java

```diff
@@ -99,6 +122,9 @@ public long transferTo(final WritableByteChannel target, final long position) th
   protected void deallocate() {
```

Comment:
In order for this code to be correct / free from leaks, we need to ensure that `deallocate()` is always eventually called once this message is constructed. Instances of this class are constructed at a single call-site in `MessageEncoder`. Once we construct the message, we immediately pass it to Netty code, which I assume will free it at the proper time.